### PR TITLE
added routs and basic navigation

### PR DESCRIPTION
**Pull request by DimaSolduridi**
added basic routing and navigation using react router dom, all routs are defind in one place, then the routs object is being passed to the router provider and the dummy page has a basic link navigation, added a custom not found page so that all incorrect urls will lead to that page and displays a 404 not found and a back button that navigates to home page

### CODE REVIEW

File: src/App.js
Comment (vy1406) [inline src/App.js:1]: Good, that's what i talked about. now add another map.
like:

```
export const ROUTES = {
    BASE: "/",
    DASHBOARD: "/dashboard",
    .....
}
```
and in myRouter do:

```
 {
        path: ROUTES.BASE,
        element: <Dummy/>,
        errorElement: <NotFound/>
    },
```
thus, you will export the routes and will use on useNavigate, 
as i said, SSOT - single source of truth, that way u can only change in ROUTES and everything will be adjusted accordingly.

File: src/pages/dummy.js
Comment (vy1406) [inline src/pages/dummy.js:1]: also here, you should change from ROUTES SSOT:

 ```
  <li>
            <Link to={ROUTES.BASE}>Home</Link>
        </li>
```
        
        and so on

File: webpack.config.js
Comment (vy1406) [inline webpack.config.js:1]: add dist folder to gitignore, should not be pushed to repo.

File: src/App.js [hunk 1]
```diff
@@ -1,7 +1,53 @@
...
+import Dashboard from "./pages/dashboard"
+import Login from "./pages/login"
+import Signup from "./pages/signup"
+import Notes from "./pages/notes"
+import Profile from "./pages/profile"
+import Todo from "./pages/todo"
+import Workouts from "./pages/workouts"
+import NotFound from './pages/notFound';
 
+const myRouter = createBrowserRouter([
```
Comment (vy1406) [inline src/App.js]: create a file routerConfig.js in root of the app ( parallel to src )

export the myRouter from there. the App.js should be clean.

**Comment by DimaSolduridi** [comment]
did all the changes you told me to do , i think it is ready to merge

**Pull request merged**